test/protocol/fundsFacet.ts

```diff
@@ -398,9 +398,39 @@ describe("Funds", function () {
           .to.be.revertedWithCustomError(fermionErrors, "UnexpectedDataReturned")
           .withArgs("0x1626ba7e000000000000000abcde000000000000000000000000000000000001");
       });
+
+      it("should use safeTransferFrom when trustedForwarder returns unexpected data", async function () {
```

Comment:
This test is a bit counter-intuitive to me - it looks more like it's testing the `safeTransferFrom` rather than `checkTrustedForwarder`.

Although nothing wrong with that, I think we should also have a test before "Revert reasons", just to show that even if `trustedForwarder` returns something weird, but the token is otherwise ok, it can still be deposited. Effectievely, the test could be the same as this, without calling `setTransferReturnData` and instead of expecting the erros, just check that the transfers happened.

And btw, this test is not passing for me
![image](https://github.com/user-attachments/assets/ac6c6986-bf19-4c79-a1f0-da0b3028f75b)